### DISCUSSION TITLE
Config for RPi 4 and a webcam

### DISCUSSION THREAD

**Discussion by mwolfeu** [bluenviron/mediamtx#2551] [Questions] [answered]
### Question

I'm trying to get my RPi to stream with a generic UVC usb webcam.  It has start up issues.  

I used the ffmpeg line in the current docs:
  cam:
    runOnInit: ffmpeg -f v4l2 -i /dev/video1 -pix_fmt yuv420p -preset ultrafast -b:v 600k -f rtsp rtsp://localhost:$RTSP_PORT/$MTX_PATH
    runOnInitRestart: yes

And when I start MediaMTX I get from ffmpeg:
closed: the stream doesn't contain any supported codec, which are currently AV1, VP9, VP8, H264, Opus, G722, G711 

I'm grateful for any pointers.  Thanks in advance!

**Comment by mwolfeu** [comment]
Also, any suggestions of working usb cams would be welcomed!

**Reply by mwolfeu** [reply, marked as answer]
I had to add -c:v h264 to the command.

**Comment by lucadici** [comment]
What is the CPU usage? Any data?

[a comment by an automated account is not shown]
